### PR TITLE
fix #488, update to JuliaLang/julia#41137

### LINKED ISSUE #488: Broken on v1.8
Especially after https://github.com/JuliaLang/julia/pull/41137.

```julia
ex = quote
    foo() = return
end
JuliaInterpreter.finish_and_return!(Frame(@__MODULE__, ex), true)
```
```
ERROR: cannot add methods to a builtin function
Stacktrace:
 [1] evaluate_methoddef(frame::Frame, node::Expr)
   @ JuliaInterpreter ~/julia/packages/JuliaInterpreter/src/interpret.jl:279
 [2] step_expr!(recurse::Any, frame::Frame, node::Any, istoplevel::Bool)
   @ JuliaInterpreter ~/julia/packages/JuliaInterpreter/src/interpret.jl:482
 [3] step_expr!
   @ ~/julia/packages/JuliaInterpreter/src/interpret.jl:591 [inlined]
 [4] finish!(recurse::Any, frame::Frame, istoplevel::Bool)
   @ JuliaInterpreter ~/julia/packages/JuliaInterpreter/src/commands.jl:14
 [5] finish_and_return!
   @ ~/julia/packages/JuliaInterpreter/src/commands.jl:30 [inlined]
 [6] finish_and_return!(frame::Frame, istoplevel::Bool)
   @ JuliaInterpreter ~/julia/packages/JuliaInterpreter/src/commands.jl:34
 [7] top-level scope
   @ none:1
```

Especially the error happens at:
https://github.com/JuliaDebug/JuliaInterpreter.jl/blob/407fa70c4b3f651fc428fca9812a91b58247759e/src/interpret.jl#L279
Clearly the `ccall` signature got staled after the PR.

I changed the line to `ccall(:jl_method_def, Cvoid, (Any, Any, Any, Any), sig, false, body, moduleof(frame))`, but had no success.
@JeffBezanson could you tell me how we should fix this ?

### PR DESCRIPTION
**Pull request by aviatesk**

### CODE REVIEW

**Comment by aviatesk** [comment]
Alright, 
```julia
        atypes = sig[1]::SimpleVector
        mt = ccall(:jl_method_table_for, Any, (Any,), Tuple{atypes...})
        ccall(:jl_method_def, Cvoid, (Any,Any,Any,Any), sig, mt, body, moduleof(frame))
```
seems to work.

I would like to make `ccall(:jl_method_def, Cvoid, (Any,Any,Any,Any), sig, nothing, body, moduleof(frame))` just work, though.

**Review by simeonschaub — APPROVED** [review]
Could you add the test case in #488 to the tests? Otherwise LGTM.

**Comment by aviatesk** [comment]
The equivalent test is already done [here](https://github.com/JuliaDebug/JuliaInterpreter.jl/blob/407fa70c4b3f651fc428fca9812a91b58247759e/test/interpret.jl#L191-L220), but would you like to add more explicit and simpler one ?

**Comment by simeonschaub** [comment]
Yeah, it's always good to have more minimal test cases as well.

**Comment by aviatesk** [comment]
Sure, done :)

File: src/interpret.jl [hunk 1]
```diff
@@ -276,7 +276,12 @@ function evaluate_methoddef(frame, node)
     length(node.args) == 1 && return f
     sig = @lookup(frame, node.args[2])::SimpleVector
     body = @lookup(frame, node.args[3])
-    ccall(:jl_method_def, Cvoid, (Any, Any, Any), sig, body, moduleof(frame))
+    # branching on https://github.com/JuliaLang/julia/pull/41137
+    @static if isdefined(Core.Compiler, :OverlayMethodTable)
+        ccall(:jl_method_def, Cvoid, (Any,Ptr{Cvoid},Any,Any), sig, C_NULL, body, moduleof(frame))
```
Comment (aviatesk) [inline src/interpret.jl:281]:
```suggestion
        ccall(:jl_method_def, Cvoid, (Any, Ptr{Cvoid}, Any, Any), sig, C_NULL, body, moduleof(frame))
```

**Review by pfitzseb — APPROVED** [review]

**Comment by aviatesk** [comment]
@pfitzseb are you going to tag a new version ?

**Comment by aviatesk** [comment]
nvm, done that already.

**Pull request merged**